recite/admin.py

```diff
@@ -55,89 +69,91 @@ class ReciterAdmin(admin.ModelAdmin):
+        # Extract audio files zip to temp directory
         audio_zip_file = form.cleaned_data['audio_zip_file']
-        temp_dir = self.get_unzipped_files_dir(audio_zip_file)
-        file_paths = self.get_file_paths(temp_dir)
-
-        # create Recitation objects from processed data
-        for row in csv_content:
-            segments_list = json.loads(row.get("segments"))
-            segments = ",".join(
-                (f"{segment[2]}:{segment[3]}" for segment in segments_list)
-            )
-
-            ayah = self.get_ayah_object_from_row(row)
-            file_ayah_path = self.get_file_audio_path(file_paths, ayah)
-
-            with open(file_ayah_path, 'rb') as f:
-                file_ayah = File(f)
-
-                recitation, _ = Recitation.objects.get_or_create(
-                    ayah=ayah,
-                    reciter=obj,
-                    segments=segments,
-                    audio=file_ayah
-                )
-
-        shutil.rmtree(temp_dir)
+        audio_files = ZipFile(audio_zip_file, 'r')
 
-    def get_file_paths(self, temp_dir):
-        """Return file paths from temp_dir checking subdirectories"""
-        file_paths = []
-        for root, directories, files in os.walk(temp_dir):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            audio_files.extractall(temp_dir)
+
+            # let's create dict file_paths[surah][ayah] = file_path
+            file_paths = self.get_file_paths_dict(temp_dir=temp_dir)
+            # let's create Recitations from segments and file_paths
+            for surah_number, ayahs in segments_dict.items():
+                for ayah_number, segments in ayahs.items():
+                    with open(file_paths[surah_number][ayah_number], 'rb')\
+                            as audio_file:
+
+                        file_ayah = File(audio_file)
+                        ayah = Ayah.objects.get(
+                            number=ayah_number,
+                            surah=surah_number,
+                        )
+
+                        Recitation.objects.get_or_create(
+                            ayah=ayah,
+                            reciter=obj,
+                            segments=segments,
+                            audio=file_ayah
+                        )
+
+    def get_file_paths_dict(self, temp_dir):
+        """
+        Process audio zip file uploaded and create dict
+        :return: file_paths[surah][ayah] = file_path
+        """
+        file_paths = defaultdict(lambda: defaultdict(int))
+
+        for root, _directories, files in os.walk(temp_dir):
             for filename in files:
-                filepath = os.path.join(root, filename)
-                file_paths.append(filepath)
+                if filename.endswith('.mp3'):
+                    filepath = os.path.join(root, filename)
+
+                    # extract surah and ayah number from filename
+                    surah_str = filename[:3]
```

Comment:
Nit: I would extract it to a separate function and test it if necessary.

Reply:
It should be standard zip file and it's single procedure of audio import done only by admins.

Reply:
I mean that lines `117-123` should probably live in a separate function.